ff/serving/provider/offline.go

```diff
@@ -89,13 +111,81 @@ func (store *memoryOfflineStore) CreateResourceTable(id ResourceID) (OfflineTabl
 }
 
 func (store *memoryOfflineStore) GetResourceTable(id ResourceID) (OfflineTable, error) {
+	return store.getMemoryResourceTable(id)
+}
+
+func (store *memoryOfflineStore) getMemoryResourceTable(id ResourceID) (*memoryOfflineTable, error) {
 	table, has := store.tables[id]
 	if !has {
 		return nil, &TableNotFound{id.Name, id.Variant}
 	}
 	return table, nil
 }
 
+// Used to implement sort.Interface for sorting.
+type materializedRecords []ResourceRecord
+
+func (recs materializedRecords) Len() int {
+	return len(recs)
+}
+
+func (recs materializedRecords) Less(i, j int) bool {
```

Comment:
I assume this is for sorting the records. Would it make sense to sort them by timestamp (given the null timestamp will always go first), since the entity identifier isn't significant with respect to ordering?

Reply:
I realize this would not necessarily guarantee deterministic ordering, so perhaps in the case of identical timestamps the entity should be the tiebreaker